lib/tests/2wp.js

```diff
@@ -175,6 +179,28 @@ const execute = (description, getRskHost) => {
             expect(batchPegoutCreatedEvent).to.not.be.null;
             expect(batchPegoutCreatedEvent.arguments.releaseRskTxHashes.includes(pegoutTransaction.transactionHash)).to.be.true;
           }
+
+          const isLovell700AlreadyActive = await Runners.common.forks.lovell700.isAlreadyActive();
+          if (isLovell700AlreadyActive) {
+            const pegoutTransactionCreatedEvent = await rskUtils.findEventInBlock(localRskTxHelper, 'pegout_transaction_created');
+            expect(pegoutTransactionCreatedEvent).to.not.be.null;
+            const encodedUtxoOutpointValues = Buffer.from(removePrefix0x(pegoutTransactionCreatedEvent.arguments.utxoOutpointValues), 'hex');
+
+            const federationUtxoValues = activeFederationUtxosBeforePegout.reduce((map, utxo) => {
+              map[utxo.valueInSatoshis] = Buffer.from(new VarInt(utxo.valueInSatoshis).encode()).toString("hex");
+              return map;
+            }, {});
+
+            let offset = 0;
```

Comment:
Too much decoding logic in a test, don't you think? What if you take the `VarInt` decoding to an utility function that returns a list of decoded outpoint values and then you can compare it like:

```javascript
const outpointValues = getDecodedOutpointsValues(encodedUtxoOutpointValues);
//...
expect(outpointValues.every(value => federationUtxoValues.has(value)).to.be.true;
```

Or something like that.

Reply:
Agree, too much stuff here not directly related to the test

Reply:
Solved